libs/cosmos-sdk/baseapp/gasuseddb.go

```diff
@@ -75,43 +78,71 @@ func (h *HistoryGasUsedRecordDB) FlushHgu() {
 	if len(h.latestGu) == 0 {
 		return
 	}
-	latestGasKeys := make([]gasKey, len(h.latestGu))
-	index := 0
-	for key, gas := range h.latestGu {
-		latestGasKeys[index] = gasKey{
-			gas: gas,
+	latestGasKeys := make([]gasKey, 0, len(h.latestGu))
+	for key, allGas := range h.latestGu {
+		latestGasKeys = append(latestGasKeys, gasKey{
+			gas: meanGas(allGas),
 			key: key,
-		}
-		index++
+		})
 		delete(h.latestGu, key)
 	}
 	h.jobQueue <- func() { h.flushHgu(latestGasKeys...) } // closure
 }
 
-func (h *HistoryGasUsedRecordDB) getHgu(key []byte) (hgu int64, fromCache bool) {
+func (h *HistoryGasUsedRecordDB) getHgu(key []byte) (hgu *HguRecord, fromCache bool) {
 	v, ok := h.cache.Get(string(key))
 	if ok {
-		return v.(int64), true
+		return v.(*HguRecord), true
 	}
 
 	data, err := h.guDB.Get(key)
 	if err != nil || len(data) == 0 {
-		return -1, false
+		return nil, false
 	}
 
-	return bytesToInt64(data), false
+	var r HguRecord
+	err = proto.Unmarshal(data, &r)
+	if err != nil {
+		return nil, false
+	}
+	return &r, false
 }
 
 func (h *HistoryGasUsedRecordDB) flushHgu(gks ...gasKey) {
 	for _, gk := range gks {
 		hgu, cacheHit := h.getHgu([]byte(gk.key))
-		// avgGas = 0.4 * newGas + 0.6 * oldGas.The value of wasm store contract is too small and need to be rounded up.
```

Comment:
0.6 is used by wasm contract, The value of wasm store contract is too small and need to be rounded up.

Reply:
deploy gas used calculate by `gc = gc/int64(deploySize) + 1`，so `0.6` is not necessary here